port/linux/omrvmem.c

```diff
@@ -1132,56 +1132,89 @@ reserve_memory_with_mmap(struct OMRPortLibrary *portLibrary, void *address, uint
 
 	Trc_PRT_vmem_reserve_entry(address, byteAmount);
 
-	if (0 != (mode & OMRPORT_VMEM_MEMORY_MODE_SHARE_FILE_OPEN)) {
+	if (OMR_ARE_ANY_BITS_SET(mode, OMRPORT_VMEM_MEMORY_MODE_SHARE_TMP_FILE_OPEN)) {
+		/* By convention, if OMRPORT_VMEM_MEMORY_MODE_SHARE_TMP_FILE_OPEN is set,
+		 * we must have the OMRPORT_VMEM_MEMORY_MODE_SHARE_FILE_OPEN bit also set.
+		 */
+		Assert_PRT_true(OMR_ARE_ANY_BITS_SET(mode, OMRPORT_VMEM_MEMORY_MODE_SHARE_FILE_OPEN));
```

Comment:
I think this should return "unsupported", consistent with other platforms.

Reply:
In my view, it's the programmer's responsibility to ensure that whenever `OMRPORT_VMEM_MEMORY_MODE_SHARE_TMP_FILE_OPEN` is set, `OMRPORT_VMEM_MEMORY_MODE_SHARE_FILE_OPEN` is also set. Thus, an assert makes more sense to me.

Reply:
In my view, it's the same sort of "programmer error" to use `OMRPORT_VMEM_MEMORY_MODE_SHARE_FILE_OPEN` on a non-Linux platform, so the response should be the same.

Reply:
Done